posthog/models/integration.py

```diff
@@ -326,7 +338,26 @@ def integration_from_oauth_response(
         config: dict = res.json()
 
         if res.status_code != 200 or not config.get("access_token"):
-            raise Exception("Oauth error")
+            # Hack to try getting sandbox auth token instead of their salesforce production account
+            if kind == "salesforce":
+                oauth_config = cls.oauth_config_for_kind("salesforce-sandbox")
+                res = requests.post(
+                    oauth_config.token_url,
+                    data={
+                        "client_id": oauth_config.client_id,
+                        "client_secret": oauth_config.client_secret,
+                        "code": params["code"],
+                        "redirect_uri": OauthIntegration.redirect_uri(kind),
+                        "grant_type": "authorization_code",
+                    },
+                )
+
+                config = res.json()
+
+                if res.status_code != 200 or not config.get("access_token"):
+                    raise Exception("Oauth error")
+            else:
+                raise Exception("Oauth error")
```

Comment:
Would you mind updating these exceptions to something more relevant? At least including the status code?

Reply:
Ideally we include the response we get, in case it has any error message.